charts/ndb-operator/templates/deployment-ndb-operator-controller-manager.yaml

```diff
@@ -0,0 +1,98 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    control-plane: controller-manager
+  name: ndb-operator-controller-manager
```

Comment:
I wanted to check with other folks, whether we need to expose the "name" field to users as well. 
e.g. name: {{ include "ndb-operator.fullname" . }}-controller-manager

This is because, we are manually generating all these YAMLs (not using helmify). Since now we have to repeat this long process for every single release going forward, the more templating logic we add, the more work needs to be done for every release in terms of updating, testing and debugging in case of issues. 

So I'd trade with caution unless having that option to our users is adding some real strong value to the product.

Reply:
That is true, but if we remove all the templating logic from everywhere then we're essentially masquerading plain yaml files as helm charts and not utilizing Helm's capabilities at all. what do others think about this? @tuxtof @yash-ntnx

Reply:
We are not removing all of the templating logic. We are retaining templating logic for namespace, operator docker image URL, replicaCount, imagePulllSecrets, resources blocks (cpu, memory) for controller manager & kubeProxy.

As discussed with @tuxtof, templating logic for the "name" is not adding much value in our use-case so was advised to remove.

Reply:
I have made the changes for templatizing the name (prefix) field. Let's discuss the need for those in the sync-up call and I can revert that commit if not necessary.